client/my-sites/preview/main.jsx

```diff
@@ -195,11 +224,17 @@ class PreviewMain extends React.Component {
 
 const mapState = state => {
 	const selectedSiteId = getSelectedSiteId( state );
+	const site = getSelectedSite( state );
+
 	return {
 		isPreviewable: isSitePreviewable( state, selectedSiteId ),
 		selectedSiteNonce: getSiteOption( state, selectedSiteId, 'frame_nonce' ) || '',
-		site: getSelectedSite( state ),
+		site: site,
 		siteId: selectedSiteId,
+
+		// TODO: This is not working correctly
+		pageOnFront: getPost( state, site.options.page_on_front ),
+		editorURL: getEditorUrl( state, site.ID, site.options.page_on_front, 'page' ),
```

Comment:
Oh is ` site.options.page_on_front` a postId?

This should be okay provided that site is populated.

```jsx
const homePagePostId = get( site, [ 'options', 'page_on_front' ] );
//...
editorUrl: getEditorUrl( state, selectedSiteId, site.options.page_on_front, 'page' ),
```